pkg/agent/agentclient/client.go

```diff
@@ -33,21 +33,33 @@ type AgentClient struct {
 	nextConnID  int64
 	connContext map[int64]*connContext
 
-	stream *RedialableAgentClient
+	stream      *RedialableAgentClient
+	serverID    string // the proxy server this client connects to
+	serverCount int    // number of proxy servers, 1 if server is not HA.
+	stopCh      <-chan struct{}
 }
 
-// NewAgentClient creates an AgentClient
-func NewAgentClient(address string, opts ...grpc.DialOption) (*AgentClient, error) {
-	stream, err := NewRedialableAgentClient(address, opts...)
+func newAgentClient(address, agentID string, cs *ClientSet, opts ...grpc.DialOption) (*AgentClient, error) {
+	stream, err := NewRedialableAgentClient(address, agentID, cs, opts...)
 	if err != nil {
 		return nil, err
 	}
+	return newAgentClientWithRedialableAgentClient(stream), nil
+}
 
-	a := &AgentClient{
+func newAgentClientWithRedialableAgentClient(rac *RedialableAgentClient) *AgentClient {
+	return &AgentClient{
 		connContext: make(map[int64]*connContext),
-		stream:      stream,
+		stream:      rac,
+		serverID:    rac.serverID,
```

Comment:
serverID, serverCount and stopCh seem to be duplicated across AgentClient and RedialableAgentClient. However the RedialableAgentClient is a field in the AgentClient. I do not believe the inner and outer struct are allowed to have different values for these fields. As such it seems like extra complexity and a source of potential bugs without providing any value.

Reply:
Good catch. Removed the fields in AgentClient.